lib/shoulda/matchers/active_record/have_secure_token_matcher.rb

```diff
@@ -0,0 +1,105 @@
+module Shoulda
+  module Matchers
+    module ActiveRecord
+      # The `have_secure_token` matcher tests usage of the
+      # `has_secure_token` macro.
+      #
+      # #### Example
+      #
+      #     class User < ActiveRecord
+      #       attr_accessor :token
+      #       attr_accessor :auth_token
+      #
+      #       has_secure_token
+      #       has_secure_token :auth_token
+      #     end
+      #
+      #     # RSpec
+      #     RSpec.describe User, type: :model do
+      #       it { should have_secure_token }
+      #       it { should have_secure_token(:auth_token) }
+      #     end
+      #
+      #     # Minitest (Shoulda)
+      #     class UserTest < ActiveSupport::TestCase
+      #       should have_secure_token
+      #       should have_secure_token(:auth_token)
+      #     end
+      #
+      # @return [HaveSecureToken]
+      #
+
+      # rubocop:disable Style/PredicateName
+      def have_secure_token(token_attribute = :token)
+        HaveSecureTokenMatcher.new(token_attribute)
+      end
+      # rubocop:enable Style/PredicateName
+
+      # @private
+      class HaveSecureTokenMatcher
+        attr_reader :token_attribute
+
+        def initialize(token_attribute)
+          @token_attribute = token_attribute
+          @issues = []
+        end
+
+        def description
+          "have :#{token_attribute} as a secure token"
+        end
+
+        def failure_message
+          "Expected #{@subject} to #{description} but the following issues " \
+          "were found: #{@issues.join(', ')}"
+        end
+
+        def failure_message_when_negated
+          "Did not expect #{@issues.join(', ')}"
+        end
+
+        def matches?(subject)
+          @subject = subject
+          @issues = []
+          add_issue_unless has_expected_instance_methods?(subject),
```

Comment:
I'm not worried necessarily about the method length. Right now the indentation makes it a little readable (at least compared to the style that's being used) and it's hard to make out what the arguments are.

I suppose, though, that it does feel a little procedural in this method. I also don't see much of a need for `add_issue_unless`, to be honest. But I like the validation pattern you've established here. So what do you think about this (suggestions for a method name are welcome)?

``` ruby
def matches?(subject)
  @subject = subject
  @errors = run_checks
  @errors.empty?
end

private

def run_checks
  errors = []

  if !has_expected_instance_methods?
    errors << "..."
  end

  if !has_expected_db_column?
    errors << "..."
  end
  
  if !has_expected_db_index?
    errors << "..."
  end

  errors
end
```